homeassistant/components/rachio/switch.py

```diff
@@ -63,32 +65,81 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+ATTR_DURATION = "duration"
+ATTR_ID = "id"
 ATTR_PERCENT = "percent"
 ATTR_SCHEDULE_SUMMARY = "Summary"
 ATTR_SCHEDULE_ENABLED = "Enabled"
 ATTR_SCHEDULE_DURATION = "Duration"
 ATTR_SCHEDULE_TYPE = "Type"
+ATTR_SORT_ORDER = "sortOrder"
 ATTR_ZONE_NUMBER = "Zone number"
 ATTR_ZONE_SHADE = "Shade"
 ATTR_ZONE_SLOPE = "Slope"
 ATTR_ZONE_SUMMARY = "Summary"
 ATTR_ZONE_TYPE = "Type"
 
+START_MULTIPLE_ZONES_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+        vol.Required(ATTR_DURATION): cv.ensure_list_csv,
+    }
+)
+
 
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the Rachio switches."""
+    zone_entities = []
     has_flex_sched = False
     entities = await hass.async_add_executor_job(_create_entities, hass, config_entry)
     for entity in entities:
+        if isinstance(entity, RachioZone):
+            zone_entities.append(entity)
         if isinstance(entity, RachioSchedule) and entity.type == SCHEDULE_TYPE_FLEX:
             has_flex_sched = True
-            break
 
     async_add_entities(entities)
     _LOGGER.info("%d Rachio switch(es) added", len(entities))
 
-    platform = entity_platform.current_platform.get()
+    def start_multiple(service):
+        """Service to start multiple zones in sequence."""
+        zones_list = []
+        person = hass.data[DOMAIN_RACHIO][config_entry.entry_id]
+        entity_id = service.data[ATTR_ENTITY_ID]
+        duration = iter(service.data[ATTR_DURATION])
+        default_time = service.data[ATTR_DURATION][0]
+
+        # Iterate through service data first to preserve zone order
+        for (count, data) in enumerate(entity_id, 1):
+            for zone in zone_entities:
+                if zone.entity_id == data:
+                    # Time can be passed as a list per zone,
+                    # or one time for all zones
+                    time = int(next(duration, default_time)) * 60
+                    zones_list.append(
+                        {
+                            ATTR_ID: zone.id,
+                            ATTR_DURATION: time,
+                            ATTR_SORT_ORDER: count,
+                        }
+                    )
+                    break
+
+        if len(zones_list) != 0:
+            person.start_multiple_zones(zones_list)
+            _LOGGER.debug("Starting zone(s) %s", entity_id)
+        else:
+            _LOGGER.warning("No matching zones found in given entity_ids")
```

Comment:
We can give direct feedback to the UI if we raise:

```suggestion
            raise HomeAssistantError("No matching zones found in given entity_ids")
```

and add
`from homeassistant.exceptions import HomeAssistantError`
above